deploy/deploy_delta_upgrade.ts

```diff
@@ -0,0 +1,208 @@
+import {HardhatRuntimeEnvironment} from "hardhat/types"
+import {DeployFunction} from "hardhat-deploy/types"
+
+import ContractDeployer from "../utils/deployer"
+import {ethers} from "hardhat"
+import {BondingManager, LivepeerGovernor, Treasury} from "../typechain"
+import getNetworkConfig from "./migrations.config"
+import {contractId} from "../utils/helpers"
+
+const PROD_NETWORKS = ["mainnet", "arbitrumMainnet"]
+
+const isProdNetwork = (name: string): boolean => {
+    return PROD_NETWORKS.indexOf(name) > -1
+}
+
+// Returns a reference to the given contract in the format used by the governor-scripts repo (e.g.
+// "ADDRESSES.arbitrumMainnet.controller"). Notice that on the serialized JSON, this will come out as a string while in
+// the governance script it should be de-stringified (remove quotes) to reference an imported ADDRESS object.
+const contractAddressRef = (network: string, name: string) => {
+    const lowerName = name[0].toLowerCase() + name.slice(1)
+    return `ADDRESSES.${network}.${lowerName}`
+}
+
+// Returns a governance action spec in the format expected by the governor-scripts to call the setContractInfo function
+// on the controller to register a contract in the protocol.
+const setContractInfoAction = (
+    network: string,
+    gitCommitHash: string,
+    name: string
+) => ({
+    target: contractAddressRef(network, "Controller"),
+    value: "0",
+    contract: "Controller",
+    name: "setContractInfo",
+    params: [contractId(name), contractAddressRef(network, name), gitCommitHash]
+})
+
+// Deploys the Livepeer Delta protocol upgrade from the previous version (962107f). This deploys only the targets
+// for already existing contracts, and skips registering them in the controller in case it's a production network.
+const func: DeployFunction = async function(hre: HardhatRuntimeEnvironment) {
+    const {deployments, getNamedAccounts} = hre // Get the deployments and getNamedAccounts which are provided by hardhat-deploy
+    const {deploy} = deployments // the deployments object itself contains the deploy function
+
+    const network = hre.network.name
+    const config = getNetworkConfig(network)
+
+    const {deployer} = await getNamedAccounts() // Fetch named accounts from hardhat.config.ts
+
+    // on prod networks, deploy contracts here but registration is done later through governance
+    // on test networks, the deployer must also be the controller owner so we can register here
+    const skipRegister = isProdNetwork(network)
+    const contractDeployer = new ContractDeployer(
+        deploy,
+        deployer,
+        deployments,
+        skipRegister
+    )
+    const controller = await contractDeployer.fetchDeployedController()
+
+    const gitCommitHash = await contractDeployer.getGitHeadCommitHash()
+    const governanceActions = []
```

Comment:
I'm also not attached to this logic to print the pending goverance actions, can easily remove it if you prefer less entropy here. It has already helped me write the initial governor script anyway.

Reply:
I'm fine with it for now. For the future, would be good to think about ways to just directly output the list of updates for the Protocol Governor in an upgrade script instead of having to print a JSON object here and copy that over into a different repo.